packages/hebao_v1/contracts/stores/QuotaStore.sol

```diff
@@ -1,27 +1,25 @@
 // SPDX-License-Identifier: Apache-2.0
 // Copyright 2017 Loopring Technology Limited.
 pragma solidity ^0.7.0;
-pragma experimental ABIEncoderV2;
 
 import "../base/DataStore.sol";
 import "../lib/MathUint.sol";
-import "../lib/Claimable.sol";
 import "../thirdparty/SafeCast.sol";
 
 /// @title QuotaStore
 /// @dev This store maintains daily spending quota for each wallet.
 ///      A rolling daily limit is used.
-contract QuotaStore is DataStore, Claimable
+contract QuotaStore is DataStore
```

Comment:
No more SLOAD optimizations by caching `Quota` once in memory and reusing it? :(

Or do we see the quota now as something that will rarely get used so no need to optimize it as much as possible?

Reply:
We should definitely optimize it, but according to Vitalik, if we write twice for the same variable, the second one only costs 200 gas, not sure if it is indeed the case. *I asked him about this in my morning*.

Reply:
Writing isn't so much the concern here, it's the reading. And unfortunately each read costs 800 gas regardless of how many times it has been read (Vitalik made a EIP to change this, but it currently doesn't have wide support so it's very uncertain if it will ever be included in a future hard fork).

Writing to the same storage slot is indeed quite cheap (well, at least cheaper than 20,000 of 5,000 gas), but the cost for this is still the same as the SLOAD cost. This was 200 gas previously, but this is now also 800 gas because of the SLOAD gas cost increase.
- https://github.com/sorpaas/EIPs/blob/c88b73889e66669db0e7094dda7907661f54d873/EIPS/eip-2200.md